src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/pages/AddPreviousName.vue

```diff
@@ -1,7 +1,156 @@
+        <EceForm ref="addPreviousNameForm" :form="previousNameForm" :form-data="formStore.formData" @updated-form-data="formStore.setFormData" />
+      </v-col>
+    </v-row>
+    <v-row class="mt-6">
+      <v-col class="d-flex flex-row ga-3 flex-wrap">
+        <v-btn size="large" color="primary" :loading="loadingStore.isLoading('profile_put')" @click="handleSubmitPreviousName">Save and continue</v-btn>
+        <v-btn size="large" variant="outlined" color="primary" @click="router.back()">Cancel</v-btn>
+      </v-col>
+    </v-row>
+  </PageContainer>
+</template>
 
 <script lang="ts">
+import { useRouter } from "vue-router";
+
+import { getProfile, putProfile } from "@/api/profile";
+import Alert from "@/components/Alert.vue";
+import EceForm from "@/components/Form.vue";
+import PageContainer from "@/components/PageContainer.vue";
+import previousNameForm from "@/config/previous-name-form";
+import { useAlertStore } from "@/store/alert";
+import { useFormStore } from "@/store/form";
+import { useLoadingStore } from "@/store/loading";
+import { useUserStore } from "@/store/user";
+import type { Components } from "@/types/openapi";
+
 export default {
   name: "AddPreviousName",
+  components: { PageContainer, EceForm, Alert },
+  setup() {
+    const formStore = useFormStore();
+    const loadingStore = useLoadingStore();
+    const userStore = useUserStore();
+    const alertStore = useAlertStore();
+    const router = useRouter();
+
+    formStore.initializeForm({});
+
+    return { formStore, loadingStore, alertStore, userStore, previousNameForm, router };
+  },
+  data: () => ({
+    items: [
+      {
+        title: "Home",
+        disabled: false,
+        to: "/",
+      },
+      {
+        title: "Profile",
+        disabled: false,
+        to: "/profile",
+      },
+      {
+        title: "Add previous name",
+      },
+    ],
+  }),
+  computed: {
+    newPreviousName(): Components.Schemas.PreviousName {
+      return {
+        firstName: this.formStore.formData[previousNameForm.inputs.firstName.id],
+        middleName: this.formStore.formData[previousNameForm.inputs.middleName.id] ?? null,
+        lastName: this.formStore.formData[previousNameForm.inputs.lastName.id],
+      };
+    },
+    duplicatePreviousName() {
+      // Check if any existing previous name matches the new previous name
+      return this.userStore.userProfile?.previousNames?.some(
+        (previousName) =>
+          previousName.firstName === this.newPreviousName.firstName &&
+          previousName.middleName === this.newPreviousName.middleName &&
+          previousName.lastName === this.newPreviousName.lastName,
+      );
+    },
+  },
+  methods: {
+    async handleSubmitPreviousName() {
+      // Validate the form
+      const { valid } = await (this.$refs.addPreviousNameForm as typeof EceForm).$refs[previousNameForm.id].validate();
+      if (valid) {
+        //check for duplicate previous name
+        if (!this.duplicatePreviousName) {
```

Comment:
Tough one here... not a big fan of nested if statements. Ideally we could put in a rule somewhere that will check for duplicates. 

in previous-name-form.ts: 
- userStore 
- formStore

then a custom rule in the input ()
- do the loop logic to see if the previous name already exists. 
- if yes, throws an error (doesn't have to give a message)